### DISCUSSION TITLE
Error while compiling with MAKE command in /opensn/build directory

### DISCUSSION THREAD

**Discussion by yingchiyu** [Open-Sn/opensn#198] [Ask for Help] [answered]
Hello, I followed the steps in install_linux.md on wsl1 (Ununtu 20.04) and I encountered a problem when compiling in the /opensn/build directory:
The result of ./configure.sh and the result of MAKE are shown in the picture, I don't know if this is related to the version of MPI or UBUNTU.
 Also, I didn't find opensn's requirement for mpi version, could you tell me if there is such a requirement?
![1](https://github.com/Open-Sn/opensn/assets/116443237/10f8f71a-2ade-4174-b305-aad8ebc73ef9)
![2](https://github.com/Open-Sn/opensn/assets/116443237/929cc0b7-fa02-4924-bef6-d559081abf1a)
![3](https://github.com/Open-Sn/opensn/assets/116443237/0a796319-441f-404b-9d24-784760ef4d1e)

I would appreciate it if you could help me!

**Comment by andrsd** [comment, marked as answer]
So, I tested the build on Ubuntu 20.04 last weekend and I got a successful build. I think what you are running into is that you used `openmpi` and I used `mpich`. So, try to change your mpi if you can. Ubuntu ships mpich 3.x and as I said that did not create any issues for me. Also, we will need to patch `mpicpp-lite` to support openmpi - we currently only test against `mpich` so we did not see this issue.

In case you want to stick to less compiling and using the packages from your distro, I used these packages from `apt`: mpich, gcc (9.4.x), lua (5.3), libreadline (8.x). I had to build PETSc and VTK myself. VTK in ubuntu 20.04 is ancient. As for PETSc, people usually  have to build it in a specific way for the application needs (OpenSn in no different). As for lua, I had to drop the requirement to 5.3 (our current check is too strict and we can lower it - I will send a PR for that this week). This is unrelated to your problem though, since you build lua yourself if i am reading the screenshots correctly.

**Reply by andrsd** [reply]
The patch for `mpicpp-lite` has been merged, see #201 .  You can try to update your `main` and recompile.

**Reply by yingchiyu** [reply]
Thank you very much for your help!! I have solved the problem as you suggested. Thank you very much!
